Automattic/wp-calypso
# Sites Dashboard: Update action's `isEligible` checks to exclude Jetpack/P2 sites from the action options

Fixes https://github.com/Automattic/dotcom-forge/issues/9971

## Proposed Changes

The Jetpack action items were lost when merging https://github.com/Automattic/wp-calypso/pull/94238/files?w=1#diff-173bec7d847197dc6a90c47661016102ba95728dab618e17b774c11c7ed1994cL550-L565, this PR adds these actions and the migration related ones back into the list. It also hides the wpcom site actions that aren't relevant to Jetpack/p2 sites.

## Testing Instructions

* [x] non-Atomic Jetpack sites
	* [x] Create a site with jurassic
	* [x] Connect your WP.com account through Jetpack
	* [x] Observe your site is listed on /sites
	* [x] Observe 'Jetpack Cloud', 'Billing', 'Support', 'Migrate to WordPress.com' actions are shown.
	* <img width="1139" alt="Screenshot 2024-11-28 at 9 46 21" src="https://github.com/user-attachments/assets/cc146a67-0b9a-4765-abd9-43ebc4099f71">
* [x] non-Atomic sites added via A4A plugin 
	* [x] Go to https://agencies.automattic.com/sites
	* [x] Click "Add sites" > "Via the Automattic plugin" and enter your site's URL
	* [x] Install and activate A4A plugin on your site 
	* [x] Connect your site
	* [x] Observe your site is listed on https://wordpress.com/sites
	* [x] Observe 'Migrate to WordPress.com' actions are shown. 
	* <img width="1140" alt="Screenshot 2024-11-28 at 9 58 28" src="https://github.com/user-attachments/assets/8d2e0a99-faa3-4c05-94ef-8bc0c0b227ab">
	* I was unsure how to create a _Jetpack disconnected_ non-Atomic site. The original code and the comment suggests _[Sites connected through A4A plugin are listed on wordpress.com/sites even when Jetpack is deactivated](https://github.com/Automattic/wp-calypso/blob/50ee79856956e06976f0e3d56b7948355af727c3/client/sites-dashboard/utils.ts#L57)_, but when you connect a site through A4A plugin, Jetpack gets connected automatically. If you disconnect Jetapck afterward, the site no longer appear on wp.com/sites anymore. 
* [x] P2 sites
	* [x] Observe some actions are hidden (See `sites-dataviews-actions.test.ts`).
* [x] Observe no regressions for Simple sites and Atomic sites
